### ISSUE TITLE
shp -> zip

### ISSUE THREAD

**Issue by rclark** [mapbox/shapefile-fairy#8]
It would be nice to teach the shapefile-fairy how to put all the shapefile bits together into a zipfile. Unfortunately [node-zipfile](https://github.com/mapbox/node-zipfile) does not create archives (as far as I can tell). Please correct me if I'm wrong @springmeyer.

**Comment by springmeyer** [comment]
Correct, it does not create archives. It could, but I've been keen to keep it's scope small. So I think this would be another case where ideally we find a pure js solution and if nothing is robust enough then consider adding to node-zipfile if its important.

Would tarring up the shapefiles (instead of zipping) suffice for what you are thinking about?

**Comment by rclark** [comment]
I'm getting to this in the context of [preprocessing](https://github.com/mapbox/preprocessorcerer). Our upload processor currently expects shapefiles to come in zipped. The **preprocessor** step is going to  have to open up the zipfile and make new, adjusted shapefiles, then submit the result back to the **processor**. If it can be resubmitted as a zipfile then the processor doesn't have to change.

But we could change that. All my previous looks for pure JS zipfile modules have come up short, but I can make the rounds again with this use-case in mind.

**Comment by springmeyer** [comment]
Ah, I see. Yeah, sounds useful/most simple to stick with zip format for this.

If we were to add node-zipfile support I'd approach this the same as the `copyFile` API in that we'd try to hide as much complexity as possible and work in files rather than in-memory buffers. Maybe something like:

```
zipfile.Create(directory,filename, function(err) { ... read filename and do something ... })
```

**Comment by rclark** [comment]
I should have a look at https://github.com/thejoshwolfe/yazl.
